src/components/page/CategorySteps.js

```diff
@@ -3,21 +3,49 @@ import { withRouter } from 'react-router'
 import connect from 'react-redux/es/connect/connect'
 import { Link } from 'react-router-dom'
 import { urlify } from '../../lib/helpers'
+import { checkPageStatus } from '../../actions/pages'
 
 class CategorySteps extends React.Component {
-  getStepsForPages(category, pages) {
-    if (!pages) {
-      return []
+  constructor(props) {
+    super(props)
+    this.state = { checkedPages: false, url: '' }
+  }
+
+  componentDidMount() {
+    this.checkPagesStatus()
+  }
+
+  componentDidUpdate() {
+    this.checkPagesStatus()
+  }
+
+  checkPagesStatus() {
+    if (!this.state.checkedPages || this.state.url !== this.props.url) {
```

Comment:
url can be retrieved with `location.href` from `react-router-dom`. There is no need to have a `prop`

Reply:
fixed